core/orch_test.go

```diff
@@ -299,30 +299,44 @@ func TestSelectTranscoder(t *testing.T) {
 	assert.NotNil(m.liveTranscoders[strm2])
 	assert.Len(m.remoteTranscoders, 2)
 
+	testSessionId := "testID"
+	testSessionId2 := "testID2"
+	testSessionId3 := "testID3"
+
 	// assert transcoder is returned from selectTranscoder
 	t1 := m.liveTranscoders[strm]
 	t2 := m.liveTranscoders[strm2]
-	currentTranscoder := m.selectTranscoder()
+	currentTranscoder, err := m.selectTranscoder(testSessionId)
+	assert.Nil(err)
 	assert.Equal(t2, currentTranscoder)
 	assert.Equal(1, t2.load)
 	assert.NotNil(m.liveTranscoders[strm])
 	assert.Len(m.remoteTranscoders, 2)
 
-	// assert transcoder with less load selected
-	currentTranscoder2 := m.selectTranscoder()
-	assert.Equal(t1, currentTranscoder2)
-	assert.Equal(1, t1.load)
+	// assert that same transcoder is selected for same sessionId
+	// and that load stays the same
+	currentTranscoder, err = m.selectTranscoder(testSessionId)
+	assert.Nil(err)
+	assert.Equal(t2, currentTranscoder)
+	assert.Equal(1, t2.load)
+	m.completeStreamSession(testSessionId)
 
-	currentTranscoder3 := m.selectTranscoder()
-	assert.Equal(t1, currentTranscoder3)
-	assert.Equal(2, t1.load)
+	// assert that a new transcoder is selected for a new sessionId
+	currentTranscoder, err = m.selectTranscoder(testSessionId2)
+	assert.Nil(err)
+	assert.Equal(t1, currentTranscoder)
+	assert.Equal(1, t1.load)
 
-	// assert no transcoder returned if all at they capacity
-	noTrans := m.selectTranscoder()
+	// Add some more load and assert no transcoder returned if all at capacity
+	currentTranscoder, err = m.selectTranscoder(testSessionId)
+	assert.Nil(err)
+	assert.Equal(t2, currentTranscoder)
```

Comment:
I don't believe these 3 Lines are required. 

t1 and t2 will already be at capacity (1) here as they both have an active session at this point. 

```suggestion

```

Reply:
Oh, I am actually closing the earlier session [here](https://github.com/reubenr0d/go-livepeer/blob/reuse-transcoder-for-stream/core/orch_test.go#L322), so this would add the load again, before asserting for "no transcoders available".